src/models/PagesEventDetails.ts

```diff
@@ -0,0 +1,78 @@
+import {PagesAnalyticsConfig} from './AnalyticsConfig';
+import {PagesEvent} from '../models';
+
+/**
+ * A fully detailed Pages Event
+ *
+ * @public
+ */
+export class PagesEventDetails implements PagesAnalyticsConfig, PagesEvent {
+  public businessId: number;
+  public eventType: string;
+  public siteId: number;
+  public production: boolean;
+  public ids?: number[];
+  public featureId: string;
+  public product: 'storepages' |'sites';
+  public path: string;
+  public pagesReferrer: string;
+  public pageType: 'entity'|'directory'|'locator'|'static';
+
+  constructor(config: PagesAnalyticsConfig, event: PagesEvent) {
+    this.eventType = event.eventType;
+    this.product = 'storepages';
```

Comment:
Is this intentionally hardcoded instead of reading from the config?

Reply:
removed this from the config as it is only ever "storepages" in the context of this library